### PR TITLE
fix a couple of privilege issues

### PR DESCRIPTION
**Pull request by nshy**
box: fix accidental exposure of _collation space to public

'public' role and thus guest and any newly created user has 'write'
privilege to _collation space. It is not enough to modify _collation as
one also needs 'create' privilege to add a collation and 'drop privilege
to delete one. So it is safe. Yet looks like we don't have any purpuse
of exposing _collation now.

box: fix unauthorized inserts into _truncate table

Non privileged user (thru public role) has write access to _truncate
table in order to be able to perform truncates on it's tables. Normally
it should be able to modify records only for the tables he has write
access. Yet now due to bootstrap check it is not so.

Closes tarantool/security#5

### CODE REVIEW

File: src/box/lua/upgrade.lua [hunk 1]
```diff
@@ -1227,6 +1227,20 @@ local function upgrade_to_2_10_1()
 end
 --------------------------------------------------------------------------------
 
+--------------------------------------------------------------------------------
+-- Tarantool 2.11.0
+--------------------------------------------------------------------------------
+local function revoke_write_access_on__collation_to_role_public()
+    local _priv = box.space[box.schema.PRIV_ID]
+    log.info("revoke write access on _collation space to public role")
```
Comment (locker) [inline src/box/lua/upgrade.lua:1235]: Nit: revoke ... from public role

File: src/box/lua/upgrade.lua [hunk 2]
```diff
@@ -1243,6 +1257,7 @@ local handlers = {
     {version = mkversion(2, 7, 1), func = upgrade_to_2_7_1, auto = true},
     {version = mkversion(2, 9, 1), func = upgrade_to_2_9_1, auto = true},
     {version = mkversion(2, 10, 1), func = upgrade_to_2_10_1, auto = true},
+    {version = mkversion(2, 11, 0), func = upgrade_to_2_11_0, auto = true},
```
Comment (locker) [inline src/box/lua/upgrade.lua:1260]: AFAIU this should be back-ported to 2.10, right? If so, you should use (2, 10, 2) here.

File: test/box-luatest/ghs_5_exclude__collation_from_public_access_test.lua [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    cg.server:stop()
+end
+
+g.test_no__collation_in_public_access = function(cg)
+    cg.server:exec(function()
+        local t = require('luatest')
+
+        -- 2 is 'public' role id
+        local r = box.space._priv:select{2, 'space', box.space._collation.id}
+        t.assert_equals(#r, 0)
```
Comment (locker) [inline test/box-luatest/ghs_5_exclude__collation_from_public_access_test.lua:20]: Please also add a test case that checks that the `_collation` space is invisible via net.box schema (`conn.space._collation` is nil).

File: src/box/lua/upgrade.lua (hunk 2 above)
Comment (nshy) [reply src/box/lua/upgrade.lua:1260]: Asked @Mons, decided not to backport.

**Review by alyapunov — APPROVED** [review]

**Review by locker — APPROVED** [review]

**Comment by locker** [comment]
Cherry-picked the first commit (about `_truncate`) to 2.10.

**Pull request merged**